### ISSUE TITLE
Multiselect component does not reload options after loading API data

### ISSUE THREAD

**Issue by lapillaga** [rizzui/rizzui#76]
Hello, I have tried to create a Multiselect component loading data from my API. The component works fine with static data, using the example in the documentation, but I have tried loading the options from my API but the problem I have is that the component does not redraw the options once the data has been loaded from the API.
This is my code
```
'use client';

import { useEffect, useState } from 'react';

import { type Table as ReactTableType } from '@tanstack/react-table';
import TableSearchInput from '@/app/shared/table/table-search-input.tsx';
import { MultiSelect, MultiSelectOption } from "rizzui";
import { useQuery } from '@tanstack/react-query';
import { getProductParentCategories } from '@/services/products-categories.tsx';

interface TableToolbarProps<T extends Record<string, any>> {
  table: ReactTableType<T>;
}

export default function TableToolbar<TData extends Record<string, any>>({
  table,
}: TableToolbarProps<TData>) {
  const [value, setValue] = useState<string[]>([]);
  const [parentCategoriesOptions, setParentCategoriesOptions] = useState<MultiSelectOption[]>([]);
  const { data: parentCategoriesData } = useQuery({
    queryKey: ['product-parent-categories'],
    queryFn: getProductParentCategories,
  });

  useEffect(() => {
    const options = parentCategoriesData?.map((category) => ({
      value: String(category.id),
      label: category.name,
    }));
    setParentCategoriesOptions(options || []);
  }, [parentCategoriesData]);

  const {
    options: { meta },
  } = table;
  
  return (
    <div className="flex w-full flex-col items-center justify-between gap-3 md:flex-row md:w-auto">
      <div className="flex w-full flex-col gap-3 md:flex-row md:w-auto">
          <MultiSelect
            searchable={true}
            className="w-full min-w-[158px] md:w-auto"
            placeholder="Categorías"
            options={parentCategoriesOptions}
            onChange={(selectedValues: string[]) => setValue(selectedValues)}
            value={value}
          />
      </div>
      <div className="w-full md:w-80">
        <TableSearchInput
          value={table.getState().globalFilter ?? ''}
          onClear={() => table.setGlobalFilter('')}
          onChange={(value) => table.setGlobalFilter(value)}
        />
      </div>
    </div>
  );
}
```
An this is the result for that
<img width="285" alt="image" src="https://github.com/user-attachments/assets/363f1cd8-c2fd-4efe-b817-0e4903ec1b1a">
So then the options are displayed when you type something in the search bar
<img width="318" alt="image" src="https://github.com/user-attachments/assets/c5084c3b-d28e-4496-ac68-a1505171e3b9">
I found an approach adding a if clause like this
```
      <div className="flex w-full flex-col gap-3 md:flex-row md:w-auto">
        { parentCategoriesOptions.length > 0 && (
          <MultiSelect
            searchable={true}
            className="w-full min-w-[158px] md:w-auto"
            placeholder="Categorías"
            options={parentCategoriesOptions}
            onChange={(selectedValues: string[]) => setValue(selectedValues)}
            value={value}
          />
        )}
      </div>
```
So this prevents the initial rendering until get my data from the backend but it's a hacky solution.
And btw is there a way to use a different type for the select options?
I use typescript, so I am forced to perform this conversion every time.
```
    const options = parentCategoriesData?.map((category) => ({
      value: String(category.id),
      label: category.name,
    }));
```

**Comment by rizaldirnm** [comment]
Its also happen on Select component

**Comment by rizaldirnm** [comment]
Looks like already on PR
https://github.com/rizzui/rizzui/pull/77 thank you so much

do you have a plan when you release it ?

**Comment by md-nahid** [comment]
We are working on some more updates. Will try to release asap.

**Comment by idris95** [comment]
@rizaldirnm We've released **rizzui@0.8.6**. Please check it out.
